lib/modules/platform/github/index.ts

```diff
@@ -705,49 +585,53 @@ function matchesState(state: string, desiredState: string): boolean {
   return state === desiredState;
 }
 
+function coerceGhRestPr(pr: GhRestPr): Pr {
```

Comment:
I think we're losing this part from GraphQL? https://github.com/renovatebot/renovate/blob/1a6feac0ea9c5157f1779bbea14342bd07160ed2/lib/modules/platform/github/index.ts#L631-L639

We need to think about the impacts. e.g. we may need to remove the "don't try to automerge if there's a negative review" and instead let github police that.

Reply:
I *suspect* that our own `git` state checking should be sufficient to detect BEHIND and CLEAN. I have no idea about HAS_HOOKS but we might be able to ignore it, and UNSTABLE means that GitHub hasn't had the chance to calculate it yet. So it might be OK that this part is removed.

Reply:
These lines are reason why I've made it draft:
https://github.com/renovatebot/renovate/blob/9bfe4f59f1c7e211325dcab510b349669f2a1634/lib/modules/platform/github/graphql.ts#L89-L93

Just wanted to research more on what exactly we need to fetch.